pkg/apis/core/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,159 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource"
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource/resourcestrategy"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Cluster defines any runtime for running containers, in the broadest
+// sense of the word "runtime".
+//
+// +k8s:openapi-gen=true
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   ClusterSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status ClusterStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// ClusterList
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Items []Cluster `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ClusterSpec defines how to find the cluster we're running
+// containers on.
+//
+// Tilt currently supports connecting to an existing Kubernetes
+// cluster or an existing Docker Daemon (for Docker Compose).
+type ClusterSpec struct {
+	// Connection spec for an existing cluster.
+	Connect *ClusterConnection `json:"connect,omitempty" protobuf:"bytes,1,opt,name=connect"`
+}
+
+// Connection spec for an existing cluster.
+type ClusterConnection struct {
+	// Defines connection to a Kubernetes cluster.
+	Kubernetes *KubernetesClusterConnection `json:"kubernetes,omitempty" protobuf:"bytes,1,opt,name=kubernetes"`
+
+	// Defines connection to a Docker daemon.
+	Docker *DockerClusterConnection `json:"docker,omitempty" protobuf:"bytes,2,opt,name=docker"`
+}
+
+type KubernetesClusterConnection struct {
+	// The name of the kubeconfig context to use.
+	//
+	// If not specified, will use the default context in the kubeconfig.
+	Context string `json:"context,omitempty" protobuf:"bytes,1,opt,name=context"`
+
+	// The default namespace to use.
+	//
+	// If not specified, will use the namespace in the kubeconfig.
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+}
+
+type DockerClusterConnection struct {
+	// The docker host to use.
+	//
+	// If not specified, will read the DOCKER_HOST env or use the default docker
+	// host.
+	Host string `json:"host,omitempty" protobuf:"bytes,1,opt,name=host"`
```

Comment:
Not critical, but I'd add a couple examples of valid Docker connection strings here - "host" is ambiguous (Docker's fault, I realize), so it'd be helpful for anyone reading the docs to realize this is something like `unix:///var/run/docker.sock` or `tcp://localhost:12345` and not just a hostname like `docker.cluster.local`

Reply:
done